mne/export/tests/test_export.py

```diff
@@ -46,6 +51,96 @@ def test_export_raw_eeglab(tmpdir):
+
+    raw.export(temp_fname)
+    if format == 'edf':
+        raw_read = read_raw_edf(temp_fname, preload=True)
+    elif format == 'bdf':
+        raw_read = read_raw_bdf(temp_fname, preload=True)
+
+    assert raw.ch_names == raw_read.ch_names
+    # only compare the original length, since extra zeros are appended
+    orig_raw_len = len(raw)
+    assert_array_almost_equal(
+        raw.get_data(), raw_read.get_data()[:, :orig_raw_len], decimal=3)
+    assert_allclose(
+        raw.times, raw_read.times[:orig_raw_len], rtol=0, atol=1e-5)
+
+
+@pytest.mark.skipif(not _check_edflib_installed(strict=False),
+                    reason='edflib-python not installed')
+@pytest.mark.parametrize(
+    ['dataset', 'format'], [
+        ['test', 'edf'],
+        ['test', 'bdf'],
+        ['misc', 'edf'],
+        ['misc', 'bdf']
+    ])
+def test_export_raw_edf_and_bdf(tmp_path, dataset, format):
+    """Test saving a Raw instance to EDF and BDF format."""
+    if dataset == 'test':
+        try:
+            import importlib_resources as imp_resrc  # py 3.7
+        except ImportError:
+            import importlib.resources as imp_resrc  # py 3.8+
+        import mne.io.tests.data
+        with imp_resrc.path(mne.io.tests.data, 'test_raw.fif') as fname:
+            raw = read_raw_fif(fname)
+    elif dataset == 'misc':
+        fname = op.join(misc.data_path(),
+                        'ecog', 'sample_ecog.edf')
+        raw = read_raw_edf(fname)
+
+    # only test with EEG channels
+    if dataset == 'misc':
+        raw = raw.set_channel_types({'EKGL': 'ecg', 'EKGR': 'ecg'})
+    raw.pick_types(eeg=True, ecog=True, seeg=True,
+                   eog=True, ecg=True, emg=True)
+    raw.load_data()
+    temp_fname = op.join(str(tmp_path), f'test.{format}')
+
+    # test runtime errors
+    with pytest.raises(RuntimeError, match='The maximum'), \
+            pytest.warns(RuntimeWarning, match='Data has a non-integer'):
+        raw.export(temp_fname, physical_range=(-1e6, 0))
+    with pytest.raises(RuntimeError, match='The minimum'), \
+            pytest.warns(RuntimeWarning, match='Data has a non-integer'):
+        raw.export(temp_fname, physical_range=(0, 1e6))
+
+    if dataset == 'test':
+        with pytest.warns(RuntimeWarning, match='Data has a non-integer'):
+            raw.export(temp_fname)
+    elif dataset == 'misc':
+        with pytest.warns(RuntimeWarning, match='EDF format requires'):
+            raw.export(temp_fname)
+
+    if 'epoc' in raw.ch_names:
+        raw.drop_channels(['epoc'])
+
+    if format == 'edf':
+        raw_read = read_raw_edf(temp_fname, preload=True)
+    elif format == 'bdf':
+        raw_read = read_raw_bdf(temp_fname, preload=True)
+
+    assert raw.ch_names == raw_read.ch_names
+    # only compare the original length, since extra zeros are appended
+    orig_raw_len = len(raw)
+    assert_array_almost_equal(
+        raw.get_data(), raw_read.get_data()[:, :orig_raw_len], decimal=3)
```

Comment:
```suggestion
        raw.get_data(), raw_read.get_data()[:, :orig_raw_len], decimal=4)
```
and also here add a comment on the choice of decimal please